tests/eval/test_2fa_duo.py

```diff
@@ -0,0 +1,408 @@
+        # Multiple CRNs: IdP displays a CRN chooser and then uses the CRN that was selected.
+        # IdP should display a CRN chooser and then uses the CRN that was selected.
+
+
+def test_remember_me_cookie(utils, sp_url, sp_domain, secrets, netid3, new_tab):
+    """
+    2FA-9 Remember me cookie
+    """
+    chrome_options = webdriver.ChromeOptions()
+    chrome_options.add_experimental_option("prefs", {"profile.default_content_settings.cookies": 1})
+    fresh_browser = Chrome(options=chrome_options)
+
+    sp = ServiceProviderInstance.diafine6
+    with utils.using_test_sp(sp):
+        fresh_browser.get(f'{sp_url(sp)}/shibevalmfa')
+        password = secrets.test_accounts.password
+        fresh_browser.send_inputs(netid3, password)
+        fresh_browser.click(Locators.submit_button)
+        fresh_browser.wait_for_tag('p', 'Use your 2FA device.')
+        fresh_browser.find_element_by_name('rememberme').click()
+        fresh_browser.switch_to.frame(fresh_browser.find_element_by_id('duo_iframe'))
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        passcode = secrets.test_accounts.duo_code
+        fresh_browser.execute_script("document.getElementsByClassName('passcode-input')[0].value=arguments[0]", passcode)
+        fresh_browser.snap()
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        fresh_browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
+        # go to an idp site to retrieve the rememberme cookie
+        fresh_browser.get('https://idp-eval.u.washington.edu/idp')
+        fresh_browser.wait_for_tag('h1', 'not found')
+
+        # look for these cookies:
+        # shib_idp_session, shib_idp_session_ss, and uw-rememberme-sptest03
+        cookie_list = fresh_browser.get_cookies()
+        wanted_cookie_names = ['shib_idp_session', 'shib_idp_session_ss', 'uw-rememberme-sptest03']
+
+        found_cookies = {}
+        for cookie in cookie_list:
+            if cookie['name'] in wanted_cookie_names:
+                found_cookies[cookie['name']] = 1
+
+        assert(len(found_cookies) == 3)
+
+        # b Sign out of the IdP to clear the 2FA session
+        fresh_browser.get('https://diafine6.sandbox.iam.s.uw.edu/Shibboleth.sso/Logout?return=https://idp-eval.u'
+                          '.washington.edu/idp/profile/Logout')
+        # After signing out of the IdP, the rememberme cookie should remain in the browser,
+        # but shib_idp_session* cookies should be gone.
+        fresh_browser.wait_for_tag('span', 'Your UW NetID sign-in session has ended.')
+
+        sign_out_cookie_list = fresh_browser.get_cookies()
+        found_rememberme = False
+        for cookie in sign_out_cookie_list:
+            if cookie['name'] == 'uw-rememberme-sptest03':
+                found_rememberme = True
+            else:
+                assert cookie['name'] not in wanted_cookie_names
+        assert found_rememberme is True
```

Comment:
This can be much more clearly expressed and easier to maintain:

```python
cookie_names = { cookie['name'] for cookie in browser.get_cookies() }
assert 'shib_idp_session' in cookie_names
assert 'shib_idp_session_ss' in cookie_names
assert 'uw-rememberme-sptest03' in cookie_names

# . . . sign out logic goes here . .  .
# Then: 

cookie_names = { cookie['name'] for cookie in browser.get_cookies() }
assert 'shib_idp_session' not in cookie_names
assert 'shib_idp_session_ss' not in cookie_names
assert 'uw-rememberme-sptest03' in cookie_names
```

Reply:
Good consolidation. I removed the length also b/c what's really important are those 3 cookies. Prod actually has 5 cookies, and the extra 2 are not of our concern.
I know they are available, but I don't think I've used a set. Curiously, why not use a list? I guess a set is faster since its implemented as a hash?